.github/workflows/bokeh-ci.yml

```diff
@@ -274,6 +276,7 @@ jobs:
           verbose: true
 
   documentation:
+    if: "startsWith(github.event.path, 'docs/') || startsWith(github.event.path, 'src/bokeh/')"
```

Comment:
On the fence about this one. There is a Sphinx directive to embed BokehJS content in docs pages that is used in the BokehJS section of the user guide, so technically a dependency here as well. But t might be OK to cut a corner here

Reply:
Moreover this job (and some other) should run also when dependencies change (i.e. `conda/` directory and perhaps other directories and files at the top-level of the repository). I'm generally not in favour of these changes. At some point we used this approach, though in a more limited manner, and to me it caused more confusion than it helped. I would be more in favour of having lightweight default workflow(s) and utilize `workflow_dispatch` for detailed/specialized (especially around releases) testsing.

Reply:
Adding `conda` as a dependency seems reasonable. 

> At some point we used this approach, though in a more

I don't recall that we every tried anything like this?

> I would be more in favour of having lightweight default workflow(s) and utilize workflow_dispatch for detailed/specialized (especially around releases) testsing.

I just don't think a reasonable default light workflow exists for everything in the repo (i.e. without this kind of filtering)

I like the idea of trying this because it makes us actually think about what the dependencies are in the repo, and be explicit about them.